internal/impl/devbox.go

```diff
@@ -634,30 +635,68 @@ func (d *Devbox) installNixProfile() (err error) {
 				_ = d.copyFlakeLockToDevboxLock()
 			}
 		}()
-	} else {
-		cmd = exec.Command(
+
+		cmd.Env = nix.DefaultEnv()
+
+		debug.Log("Running command: %s\n", cmd.Args)
+		_, err = cmd.Output()
+
+		var exitErr *exec.ExitError
+		if errors.As(err, &exitErr) {
+			return errors.Errorf("running command %s: exit status %d with command stderr: %s",
+				cmd, exitErr.ExitCode(), string(exitErr.Stderr))
+		}
+		if err != nil {
+			return errors.Errorf("running command %s: %v", cmd, err)
+		}
+
+		return nil
+	}
+
+	// Non flakes below:
+	packages := append([]string{""}, d.cfg.Packages...)
+
+	total := len(packages)
+	for idx, pkg := range packages {
+		stepNum := idx + 1
+
+		info, infoFound := nix.PkgInfo(d.cfg.Nixpkgs.Commit, pkg)
+		var msg string
+		if pkg == "" {
+			msg = fmt.Sprintf("[%d/%d] nixpkgs", stepNum, total)
+		} else if infoFound {
+			msg = fmt.Sprintf("[%d/%d] %s (%s)", stepNum, total, pkg, info.Version)
+		} else {
+			msg = fmt.Sprintf("[%d/%d] %s", stepNum, total, pkg)
+		}
+
+		step := stepper.Start(msg)
+
+		cmd := exec.Command(
 			"nix-env",
 			"--profile", profileDir,
+			// TODO savil. hook this up to gcurtis's mirrorURL
+			"-f", fmt.Sprintf("https://github.com/nixos/nixpkgs/archive/%s.tar.gz", d.cfg.Nixpkgs.Commit),
 			"--install",
-			"-f", filepath.Join(d.projectDir, ".devbox/gen/development.nix"),
 		)
-	}
-
-	cmd.Env = nix.DefaultEnv()
-
-	debug.Log("Running command: %s\n", cmd.Args)
-	_, err = cmd.Output()
+		if pkg != "" {
+			cmd.Args = append(cmd.Args, "--attr", pkg)
+		} else {
+			// TODO savil. Figure out how to pre-install just the nixpkgs
+			step.Fail(msg)
```

Comment:
If you're looking to make a step that just warms the nixpkgs cache and nothing else, you can try:

```
nix-instantiate --eval --expr path https://github.com/nixos/nixpkgs/archive/af9e00071d0971eb292fd5abef334e66eda3cb69.tar.gz
```

Reply:
yes! thanks. Let me try this....

Reply:
@gcurtis hmm what does one put for `--expr path` above?